content/_data/changelogs/_STYLEGUIDE.adoc

```diff
@@ -0,0 +1,56 @@
+= Jenkins Changelog Styleguide
+
+The Changelog describes notable changes in Jenkins.
+Not all changes are included, but all changes impacting users or developers are.
+
+== Types of changes
+
+The changelog supports five different types of changes:
+
+* *security* refers specifically to security fixes.
+  These changes are typically just references to a security advisory.
+* *major bug* are major bug fixes. *bug* are all other bug fixes.
+* *major rfe* are major new features and improvements. *rfe* are all other features and improvements.
+
+Library upgrades are generally assigned the label matching the motivation for the upgrade, which should be reflected in the label:
+_Upgraded foo from 1.2 to 1.4 to fix problems related to widget frobnication._
+
+When in doubt about the nature of a change, a change should be marked as an RFE and not a bug fix.
+
+Use the *major* label for changes that are very important, or have a high risk of introducing regressions (which often corresponds to the number of lines changed).
+
+== Labels
+
+* Use present tense to describe the specific change. "Fix a problem", rather than "Fixed a problem".
+  Descriptions of previous (now obsolete) behavior can deviate from this rule.
+* Use the "Developer:" prefix for any changes expected to only matter to developers of core or plugins.
+* Use the "Internal:" prefix for any notable internal changes with no expected impact outside core.
+* When upgrading a library, include both the former and current version number.
+* Use complete sentences as label, and end them with a period.
+** Changelog entries related to security fixes are an exception to this, as they only state _Security fixes_ or _Important security fixes_.
+   Use the latter label in case any fixes are considered high severity.
+
+== Messages
+
+Up to one highlighted message can be assigned to a Jenkins release.
+Reserve these messages for important announcements that go beyond regular changes.
+Examples include new required minimum Java versions, introduction of a system like JEP-200 with expected widespread impact, or when a release is very broken (e.g. imprope
+
```

Comment:
"Links" section is missing in the guide. It's important to clarify it IMHO (e.g. referencing PRs/Issues, linking upstream changelogs, etc.)

Reply:
What would you add there? That we should reference an upstream changelog, if available, when updating a component?

Note that the changelog format already has support for referencing issues and PRs, don't want to rehash that here.